File: BaekJoon/Sorting/18870.py
import sys

N = int(sys.stdin.readline())

li = list(map(int, sys.stdin.readline().split()))

temp = sorted(list(set(li)))

count = {}

i = 0
while i < len(temp):
    if temp[i] not in count.keys():
        count[temp[i]] = i
    i+=1

for num in li:
    print(count[num], end =" ")

File: BaekJoonSwift/BaekJoonSwift/Sorting/18870.swift
import Foundation

let n = Int(readLine()!)!
let arr = readLine()!.split(separator: " ").map { Int($0)! }
var dict = [Int: Int]()
var temp = 0
var result = ""

for i in arr.sorted() {
    if dict[i] == nil {
        dict[i] = temp
        temp += 1
    }
}

arr.forEach {
    result += "\(dict[$0]!) "
}

print(result)

